plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -372,14 +372,17 @@ def grouped_deploy_group_roles
 
     # updates resources via kubernetes api
     def deploy(release_docs)
-      release_docs.each do |release_doc|
+      release_docs.map do |release_doc|
         puts_action "Deploying", release_doc
-        if release_doc.blue_green_color
-          non_service_resources(release_doc).each(&:deploy)
-        else
-          release_doc.deploy
+
+        Thread.new do
+          if release_doc.blue_green_color
```

Comment:
prefer to map first and then threaded `.deploy` to separate concerns a bit better

Reply:
also use `Parallel.map` so exceptions are handled (we don't need to DB multithreading that Samson::Parallelizer provides)